birdhouse/components/weaver/post-docker-compose-up

```diff
@@ -0,0 +1,149 @@
+    cookie_jar=$( \
+        curl -k -s -m 2 -o /dev/null \
+             -X POST \
+             -H "Content-Type: application/json" \
+             -d "{\"user_name\": \"${MAGPIE_ADMIN_USERNAME}\", \"password\": \"${MAGPIE_ADMIN_PASSWORD}\"}" \
+             --cookie-jar - \
+             "${MAGPIE_URL}/signin" \
+    )
+    # trim excess stuff in cookie_jar pseudo-file (comments, empty lines)
+    cookie_jar=$(echo "${cookie_jar}" | grep -v '# ' | grep -v -e '^$')
+    # validate exactly 1 cookie retrieved (empty if bad-auth or invalid endpoint)
+    if [ "$(echo "${cookie_jar}" | wc -l)" -eq 1 ]; then
+        fields=$(echo "${cookie_jar}" | wc -w)
+        cookie_name=$(echo "${cookie_jar}" | cut -f $(( fields - 1 )) )
+        cookie_value=$(echo "${cookie_jar}" | cut -f "${fields}")
+        cookie="${cookie_name}=${cookie_value}"
+        break;
+    fi
+    # interrupt if max time reached
+    next_time=$(date -u +%s)
+    delta_time=$(( next_time - start_time ))
+    if [ ${delta_time} -ge "${WEAVER_WPS_PROVIDERS_MAX_TIME}" ]; then
+        msg="Failed to register all provider specified in WEAVER_WPS_PROVIDERS."
+        printf "%s" "${ERROR}Timeout (${WEAVER_WPS_PROVIDERS_MAX_TIME}s)! ${msg}"
+        exit 1
+    fi
+done
+if [ -z "${cookie}" ]; then
+    echo "${ERROR}Failed to retrieve authentication token from Magpie for Weaver WPS providers registration."
+    exit 2
+fi
+
+# parse providers
+WEAVER_WPS_PROVIDERS="$(echo "${WEAVER_WPS_PROVIDERS}" | tr ',' ' ')"
+
+# move on to actual registration of WPS providers
+echo "${PREFIX}Using URL: [${WEAVER_URL}]"
+start_time="$(date -u +%s)"
+ret=1
+for prov in ${WEAVER_WPS_PROVIDERS}; do
+    if [ -z "${prov}" ]; then
+        continue
+    fi
+    prov_url="https://${PAVICS_FQDN_PUBLIC}${TWITCHER_PROTECTED_PATH}/${prov}"
+    prov_cap="${prov_url}?service=WPS&request=GetCapabilities"
+
+    # wait for WPS provider to respond
+    printf "%s" "${PREFIX}Wait for response from remote WPS provider [${prov}] on [${prov_url}]."
+    while true; do
+        # request the URL and obtain the body+http code, then split them for verification
+        resp=$(curl -k -s -m 2 -w "\n%{http_code}" -b "${cookie}" "${prov_cap}")
+        ret=$?   # in case not up yet
+        code=$(echo "${resp}" | tail -n -1)
+        body=$(echo "${resp}" | head -n -1)
+        caps=$(echo "${body}" | grep -c "wps:Capabilities")
+        if [ ${ret} -eq 0 ] && [ "${code}" -eq 200 ] && [ "${caps}" -ne 0 ]; then
+            printf "\n%s\n" "${PREFIX}Got valid response from remote WPS provider [${prov}]."
+            break;
+        fi
+        sleep 1
+        printf "."
+
+        # interrupt if max time reached
+        next_time=$(date -u +%s)
+        delta_time=$(( next_time - start_time ))
+        if [ ${delta_time} -ge "${WEAVER_WPS_PROVIDERS_MAX_TIME}" ]; then
+            msg="Failed to register all provider specified in WEAVER_WPS_PROVIDERS."
```

Comment:
You are waiting on each provider and if ready, add each of them right away so this message is misleading.  Mention the current provider that failed to be ready instead.